### PR TITLE
Add a way to only log errors after a number of errors in a row

### PR DESCRIPTION
**Pull request by lahabana**
When running multiple instances it is expected to get than transactions to fail.
  In the current state, it was annoying because there would be errors in the log for
  something that's expected to happen.
  We've now introduced a number of errors in a row to have before actually logging an error.
  The default behaviour is unchanged but users that are annoyed by error logs can reduce the
  likeliness by increasing this.A

### CODE REVIEW

**Comment by lahabana** [comment]
@roblaszczak let me know if you don't agree with this approach!
The motivation behind this change is that we have pretty frequent spikes of errors: `"could not unmarshal message from query: could not scan message row: pq: could not serialize access due to concurrent update"` and we like to keep our error log quite clean.

Another way to go about it was to grab the serialization errors from the drivers to return a "retriable error" but this looked a lot trickier.

**Comment by lahabana** [comment]
@roblaszczak I'd be interested to hear if you have another suggestion to avoid the logging noise. It makes our services a little harder to monitor

**Comment by m110** [comment]
Hey @lahabana I'm wondering if the transaction errors have some specific codes that we could check, like we do with `isDeadlock`?

**Review by m110 — APPROVED** [review]

File: pkg/sql/subscriber.go [hunk 1]
```diff
@@ -191,23 +199,36 @@ func (s *Subscriber) consume(ctx context.Context, topic string, out chan *messag
...
 					"message_uuid": messageUUID,
 				})
 			} else {
-				logger.Error("Error querying for message", err, watermill.LogFields{
-					"wait_time": s.config.RetryInterval,
-				})
+				if errorsInARow >= s.config.ErrorLogThreshold {
+					logger.Error("Error querying for message, will retry", err, watermill.LogFields{
+						"wait_time":    s.config.RetryInterval,
+						"errorsInARow": errorsInARow,
```
Comment (m110) [inline pkg/sql/subscriber.go:212]: Perhaps we could use `errors_in_a_row` just to be consistent with snake_case in field names?

Also worth considering if we could deduplicate the message and fields a bit maybe.

**Comment by lahabana** [comment]
@m110 Thanks for the review. For the error it's tricky as MySQL and Postgres have different Isolation properties.
I think in practice and [this article](https://postgresql.verite.pro/blog/2020/02/14/isolation-repeatable-read-postgresql-mysql.html) seems to confirm it, MySQL doesn't error when using `repeatable Reads`.
Therefore I think we could do a hack similar to what is done for the deadlock with `could not serialize` would work as [Mysql](https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html) doesn't have an error that would match.

I'll update this with enough comments :) to explain this

**Comment by lahabana** [comment]
Replacing this with #15 as it seems like a better way to solve this problem

**Pull request closed without merging**